src/sentry/models/pullrequest.py

```diff
@@ -1,11 +1,49 @@
+from __future__ import annotations
+
+from typing import Any, Mapping
+
 from django.db import models
 from django.db.models.signals import post_save
 from django.utils import timezone
 
-from sentry.db.models import BoundedPositiveIntegerField, FlexibleForeignKey, Model, sane_repr
+from sentry.db.models import (
+    BaseManager,
+    BoundedPositiveIntegerField,
+    FlexibleForeignKey,
+    Model,
+    sane_repr,
+)
 from sentry.utils.groupreference import find_referenced_groups
 
 
+class PullRequestManager(BaseManager):
+    def update_or_create(
+        self,
+        defaults: Mapping[str, Any] | None = None,
+        **kwargs: Any,
+    ) -> tuple[Model, bool]:
+        """
+        Wraps `update_or_create()` and ensures `post_save` signals are fired for
+        updated records as `GroupLink` functionality is dependent on signals
+        being fired.
+        """
+        organization_id = kwargs.pop("organization_id")
+        repository_id = kwargs.pop("repository_id")
+        key = kwargs.pop("key")
+
+        affected, created = super().update_or_create(
```

Comment:
Curious what others think about this change, but I don't mind overwriting the existing `update_or_create` method from django if we're still calling it with `super()` later on. The included docstring should inform anyone using it that it's not the django authored version of the method.